pkg/agent/agent.go

```diff
@@ -67,18 +69,42 @@ type proc interface {
 	KillProcess() error
 }
 
-func New(log logging.Logger, kube kubernetes.Interface, plat platform.Platform, nodeName string) (*Agent, error) {
+func New(log logging.Logger, kube kubernetes.Interface, platform platform.Platform, nodeName string) (*Agent, error) {
```

Comment:
Please correct me if I'm wrong @jahkeup 

>     * What's the expected behavior when `kube == nil`?

I believe this is so that we can test the agent locally.



> * Why do we both accept `platform` as a parameter and then also try to determine it by querying with the `nodeclient`?

For testing purposes. As pointed out by https://github.com/bottlerocket-os/bottlerocket-update-operator/pull/35#discussion_r453110900, it should probably be removed. I'll go ahead and do that.

Reply:
Yes, `kube == nil` is only expected and used in test cases.